components/esp_common/include/esp_pm.h

```diff
@@ -175,6 +180,13 @@ esp_err_t esp_pm_lock_delete(esp_pm_lock_handle_t handle);
  */
 esp_err_t esp_pm_dump_locks(FILE* stream);
 
+#if CONFIG_FREERTOS_USE_TICKLESS_IDLE
+/**
+ * Set an optional callback function to be invoked on wakeup from automatic light sleep.
+ */
+void esp_pm_set_light_sleep_wakeup_cb(esp_pm_light_sleep_wakeup_callback cb);
```

Comment:
Have you considered adding the callback as a new member to the `esp_pm_config_t` structure? This would eliminate the need for adding `esp_pm_set_light_sleep_wakeup_cb`.

Reply:
could you please also add a sentence to the description of this callback, explaining that no blocking functions can be called from the callback, including printf/ESP_LOG.

Reply:
I'm wary about fiddling with the struct as it may be breaking (in my understanding, at least).
eg. If currently using a local esp_pm_config_esp32_t variable, when the struct definition is updated with a new pointer this field will be uninitialised, causing esp_pm_configure to set the callback to an undefined address (if not null). I'll leave it to someone else to decide if that's acceptable.